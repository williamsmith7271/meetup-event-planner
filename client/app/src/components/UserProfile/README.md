## UserProfile Component
A component that represents the user's profile, allowing the user to see and interact with data.

### Example

```js
<UserProfile
  user={user}
  onEditBio={mockFn}
  onCancel={mockFn}
  onSaveEdit={mockFn}
  onClickToEdit={mockFn}
  onEditEmail={mockFn}
  onEditAvatar={mockFn}
  onEditEmail={mockFn}
  onEditEmployer={mockFn}
  isEditing={false}
  bioInput={''}
  avatarInput={''}
  emailInput={''}
  employerInput={''}
/>
```

### Props
See index.js
