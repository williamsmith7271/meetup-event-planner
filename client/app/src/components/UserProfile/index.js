import React, { PropTypes } from 'react';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
import { EditableField } from 'components';
import Section from 'grommet-udacity/components/Section';
import Article from 'grommet-udacity/components/Article';
import Heading from 'grommet-udacity/components/Heading';
import Paragraph from 'grommet-udacity/components/Paragraph';
import Box from 'grommet-udacity/components/Box';
import Button from 'grommet-udacity/components/Button';
import List from 'grommet-udacity/components/List';
import ListItem from 'grommet-udacity/components/ListItem';
import Footer from 'grommet-udacity/components/Footer';
import Menu from 'grommet-udacity/components/Menu';
import Anchor from 'grommet-udacity/components/Anchor';

const UserProfile = ({
  user,
  onEditBio,
  isEditing,
  bioInput,
  onSaveEdit,
  onClickToEdit,
  onCancel,
  onEditAvatar,
  avatarInput,
  onEditEmail,
  emailInput,
  onEditEmployer,
  employerInput,
}) => (
  <Section className={styles.userProfile}>
    <Box
      justify="center"
      align="center"
      pad={{ horizontal: 'large', vertical: 'large' }}
    >
      <Article className={styles.panel}>
        <Heading tag="h2" align="center">
          {`Hello, ${user.name}!`}
        </Heading>
        <Section
          className={isEditing ? '' : styles.transformAvatar}
          pad={{ vertical: 'medium' }}
          align="center"
          justify="center"
        >
          <EditableField
            onClickToEdit={onClickToEdit}
            isEditing={isEditing}
            onEdit={onEditAvatar}
            value={avatarInput || user.avatar}
            name="avatar"
          >
            <img
              className={styles.avatar}
              src={user.avatar ?
                user.avatar
              :
                'http://bit.ly/2dqCGdd'
              }
            />
          </EditableField>
        </Section>
        <EditableField
          isEditing={isEditing}
          onClickToEdit={onClickToEdit}
          onEdit={onEditBio}
          value={bioInput || user.bio}
          name="bio"
        >
          <Paragraph className={`${styles.isButton} ${styles.paragraph}`}>
            {user.bio ? user.bio : 'Click to add a bio.'}
          </Paragraph>
        </EditableField>
        <EditableField
          onClickToEdit={onClickToEdit}
          isEditing={isEditing}
          onEdit={onEditEmployer}
          value={employerInput || user.employer}
          name="employer"
        >
          <Paragraph className={styles.isButton}>
            {user.employer ? user.employer : 'Click to add a employer.'}
          </Paragraph>
        </EditableField>
        <EditableField
          isEditing={isEditing}
          onEdit={onEditEmail}
          onClickToEdit={onClickToEdit}
          value={emailInput || user.email}
          name="email"
        >
          <Heading
            className={styles.isButton}
            tag="h3"
            align="center"
          >
            {user.email}
          </Heading>
        </EditableField>
        {isEditing &&
          <Footer align="center" justify="center">
            <Menu inline direction="row" responsive={false}>
              <Button
                label="Save"
                style={{ marginTop: 10 }}
                onClick={onSaveEdit}
              />
              <Button
                label="Cancel"
                style={{ marginTop: 10 }}
                onClick={onCancel}
              />
            </Menu>
          </Footer>
        }
      </Article>
      <Article className={styles.panel}>
        <Section pad={{ vertical: 'medium' }} align="center" justify="center">
          <Heading tag="h2">
            Event History
          </Heading>
          {user.events && user.events.length > 0 ?
            <List>
              {user.events.map((item, i) =>
                <ListItem key={i}>
                  <Anchor href={`/events/${item.id}`}>
                    {item.name}
                  </Anchor>
                </ListItem>
              )}
            </List>
          :
            <Box>
              <Heading tag="h3">
                No Events Created Yet
              </Heading>
              <Button label="Create One" href="/create-event" />
            </Box>
          }
        </Section>
      </Article>
    </Box>
  </Section>
);

UserProfile.propTypes = {
  onEditEmail: PropTypes.func.isRequired,
  isEditing: PropTypes.bool.isRequired,
  emailInput: PropTypes.string.isRequired,
  onEditBio: PropTypes.func.isRequired,
  onSaveEdit: PropTypes.func.isRequired,
  onClickToEdit: PropTypes.func.isRequired,
  onCancel: PropTypes.func.isRequired,
  onEditAvatar: PropTypes.func.isRequired,
  avatarInput: PropTypes.string,
};

export default cssModules(UserProfile, styles);
