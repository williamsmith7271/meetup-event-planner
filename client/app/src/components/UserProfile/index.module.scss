.avatar {
  width: 100px;
  height: 100px;
  border-radius: 50%;
  border: 2px solid transparent;
  overflow: hidden;
  cursor: pointer;
  transition: all .3s ease-in-out;
  border-radius: 50%;
  border: 1px solid #dbe2e8;
  margin: 0 auto;
  position: relative;
}

.input {
  width: 100%;
}

.panel {
  background: #fff;
  box-shadow: 0 2px 4px 0 rgba(167, 157, 144, 0.52);
  width: 100%;
  margin-bottom: 80px;
  position: relative;
  padding: 60px 0 !important;
}

.transformAvatar {
  display: inline-block;
  line-height: 1;
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  top: -100px;
  left: 0;
  width: 100%;
}
