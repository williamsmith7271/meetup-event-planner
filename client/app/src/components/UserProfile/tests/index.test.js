import UserProfile from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<UserProfile />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <UserProfile />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
