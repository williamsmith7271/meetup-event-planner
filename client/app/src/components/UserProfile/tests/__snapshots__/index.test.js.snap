exports[`<UserProfile /> should render with default props 1`] = `
<Section
  className={undefined}
  pad={
    Object {
      "vertical": "medium"
    }
  }>
  <Box
    align="center"
    announce={false}
    direction="column"
    focusable={true}
    justify="center"
    pad={
      Object {
        "horizontal": "large",
        "vertical": "large"
      }
    }
    responsive={true}
    tag="div">
    <Article
      className={undefined}
      direction="column"
      pad="none">
      <Heading
        align="center"
        tag="h2">
        Hello, undefined!
      </Heading>
      <Section
        align="center"
        className={undefined}
        justify="center"
        pad="none">
        <undefined
          altName="avatar"
          autoFocus={true}
          isEditing={false}
          name="photo"
          onClickToEdit={[Function mockFn]}
          onEdit={[Function mockFn]}
          placeholder="http://github.com/avatar.png"
          type="url"
          value="">
          <img
            className={undefined}
            src="http://bit.ly/2dqCGdd" />
        </undefined>
      </Section>
      <undefined
        isEditing={false}
        name="bio"
        onClickToEdit={[Function mockFn]}
        onEdit={[Function mockFn]}
        placeholder="Hi, I\'m Bill and I used to be the president."
        value="">
        <Paragraph
          className="undefined undefined">
          Click to add a bio.
        </Paragraph>
      </undefined>
      <undefined
        altName="employer"
        isEditing={false}
        name="organization"
        onClickToEdit={[Function mockFn]}
        onEdit={[Function mockFn]}
        placeholder="Udacity"
        value="">
        <Paragraph
          className={undefined}>
          Click to add a employer.
        </Paragraph>
      </undefined>
      <undefined
        isEditing={false}
        name="email"
        onClickToEdit={[Function mockFn]}
        onEdit={[Function mockFn]}
        placeholder="bill@clinton.com"
        type="email"
        value="">
        <Heading
          align="center"
          className={undefined}
          tag="h3" />
      </undefined>
    </Article>
    <Article
      className={undefined}
      direction="column"
      pad="none">
      <Section
        align="center"
        justify="center"
        pad={
          Object {
            "vertical": "medium"
          }
        }>
        <Heading
          tag="h2">
          Event History
        </Heading>
        <Box
          announce={false}
          direction="column"
          focusable={true}
          pad="none"
          responsive={true}
          tag="div">
          <Heading
            tag="h3">
            No Events Created Yet
          </Heading>
          <Button
            href="/create-event"
            label="Create One"
            type="button" />
        </Box>
      </Section>
    </Article>
  </Box>
</Section>
`;
